admin/tools/docker/replication/deployment/ncsa/env.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+
+# This file is part of lsst/qserv.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+# This script is supposed to be sourced from a client script in order
+# to set up proper values of the corresponding parameters
+
+basedir=$(dirname $0)
+if [ -z "$basedir" ] || [ "$0" = "bash" ]; then
+    (>&2 echo "error: variable 'basedir' is not defined")
+    exit 1 
+fi
+basedir="$(readlink -e $basedir)"
+if [ ! -d "$basedir" ]; then
+    (>&2 echo "error: path 'basedir' is not a valid directory")
+    exit 1
+fi
+
+function get_param {
```

Comment:
In general I'd avoid defining functions (and variables) with common names in sourced script. Also this function is more or less equivalent to `cat "$basedir/$1"`, its error handling is not much different from error handling of `cat` command itself. Just use `cat` instead of this function.

Reply:
Noted. Though, I would still prefer to have the function as it's more descriptive then just 'cat'. What I'm going to do instead is to add the following line in the end of this script to get rid of the function to avoid "contaminating" the namespace:
`unset -f get_param`